ui/lib/pki/addon/components/pki-generate-csr.ts

```diff
@@ -0,0 +1,68 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { task } from 'ember-concurrency';
+import { waitFor } from '@ember/test-waiters';
+import { expandAttributeMeta } from 'vault/utils/field-to-attrs';
+import FlashMessageService from 'vault/services/flash-messages';
+import PkiActionModel from 'vault/models/pki/action';
+import errorMessage from 'vault/utils/error-message';
+
+interface Args {
+  model: PkiActionModel;
+  useIssuer: boolean;
+  onSave: CallableFunction;
+  onCancel: CallableFunction;
+}
+
+export default class PkiGenerateIntermediateComponent extends Component<Args> {
+  @service declare readonly flashMessages: FlashMessageService;
+
+  @tracked modelValidations = null;
+  @tracked error: string | null = null;
+  @tracked alert: string | null = null;
+
+  formFields;
+
+  constructor(owner: unknown, args: Args) {
+    super(owner, args);
+    this.formFields = expandAttributeMeta(this.args.model, [
+      'type',
+      'commonName',
+      'excludeCnFromSans',
+      'format',
+      'serialNumber',
+      'addBasicConstraints',
+    ]);
+  }
+
+  @action
+  cancel() {
+    this.args.model.unloadRecord();
+    this.args.onCancel();
+  }
+
+  @task
+  @waitFor
+  *save(event: Event) {
+    event.preventDefault();
+    try {
+      const { model, onSave } = this.args;
+      const { isValid, state, invalidFormMessage } = model.validate();
+      if (isValid) {
+        yield model.save({
+          adapterOptions: { actionType: 'generate-csr', useIssuer: model.canGenerateIssuerIntermediate },
```

Comment:
Since the model uses `lazyCapabilities` I found that if we only call it during the `save` action it doesn't always get the response back in time to be useful to this method. So I recommend calculating it as you pass in `useIssuer` and then replace this with that arg. That way the parent can easily override it as well. 
```suggestion
          adapterOptions: { actionType: 'generate-csr', useIssuer: this.args.useIssuer },
```

Reply:
I'm glad you pointed that out! I was working on tests for the component and noticed that it wasn't picking up the capabilities but it was when I went through the same steps in the app.

I thought we wanted to get away from evaluating form validation on change for UX purposes since the errors show up immediately and can be distracting. I can make the change but since `generateIssuerCsrPath` is a promise proxy, what if I were to await that before calling save?

```js
const issuerCapabilities = yield model.generateIssuerCsrPath;
yield model.save({
  adapterOptions: { actionType: 'generate-csr', useIssuer: issuerCapabilities.get('canCreate') }
});
```
With the above change the test is now passing!

Reply:
Interesting. I don't love having to do all the manual jumps to get it to Just Work, but it's good to know that yielding the path manually does work! One other thing to note is that if we go with your proposed code, that does differ from the patterns we have already established in the other PKI forms. If we go with what you propose I would ask that we remove `useIssuer` from the component args and maybe document this usage in the `lazyCapabilities` definition?

Reply:
Sorry I enabled auto merge since I stepped out to walk the dog. I reverted those changes and my test is failing again. I updated the `pki-generate-csr` to pass in `adapterOptions` instead and the capabilities are resolving after the save request. The same is true if I pass just the property as `useIssuer` as suggested. Even though `canGenerateIssuerIntermediate` is passed in as an arg the computed is not triggered until the property is accessed. I suspect a similar test would fail in the generate-root component.

Reply:
The `useIssuer` arg in the interface should have been removed since it was from an early iteration of the component so I will get that updated. I can add a test for saving the config from the generate-root component and see if it fails and then align the capabilities access between the 2 components. I can also add some comments somewhere to note that we should be waiting for the promises if not accessed from a template.